integration/spark/src/main/common/java/io/openlineage/spark/agent/facets/builder/ErrorFacetBuilder.java

```diff
@@ -0,0 +1,26 @@
+package io.openlineage.spark.agent.facets.builder;
+
+import io.openlineage.client.OpenLineage.RunFacet;
+import io.openlineage.spark.agent.facets.ErrorFacet;
+import io.openlineage.spark.api.CustomFacetBuilder;
+import java.util.function.BiConsumer;
+import org.apache.spark.scheduler.JobFailed;
+import org.apache.spark.scheduler.SparkListenerJobEnd;
+
+/** {@link CustomFacetBuilder} that generates an {@link ErrorFacet} for job failure events. */
+public class ErrorFacetBuilder extends CustomFacetBuilder<SparkListenerJobEnd, RunFacet> {
+
+  @Override
+  public boolean isDefinedAt(Object x) {
+    return super.isDefinedAt(x) && ((SparkListenerJobEnd) x).jobResult() instanceof JobFailed;
+  }
+
+  @Override
+  protected void build(SparkListenerJobEnd event, BiConsumer<String, ? super RunFacet> consumer) {
+    if (event.jobResult() instanceof JobFailed) {
+      consumer.accept(
+          "spark.exception",
```

Comment:
Minor: I think we should be using underscores (`_`) as defined by the [spec](https://github.com/OpenLineage/OpenLineage/blob/main/spec/OpenLineage.md#custom-facet-naming)?

```
spark_exception
```

Reply:
Cool. I would do the name change as a follow up PR.